packages/ingest/granule.js

```diff
@@ -235,37 +290,64 @@ class Granule {
     }
   }
 
-  async _cksum(tempFile) {
+  /**
+   * Get cksum checksum value of file
+   *
+   * @param {string} filepath - filepath of file to checksum
+   * @returns {Promise} checksum value calculated from file
+   */
+  async _cksum(filepath) {
     return new Promise((resolve, reject) =>
-      fs.createReadStream(tempFile)
+      fs.createReadStream(filepath)
         .pipe(cksum.stream((value) => resolve(value.readUInt32BE(0))))
-        .on('error', reject)
-    );
+        .on('error', reject));
   }
 
-  async _hash(type, tempFile) {
-    const options = { algorithm: type };
+  /**
+  * Get hash of file
+  *
+  * @param {string} algorithm - algorithm to use for hash,
+  * any algorithm accepted by node's `crypto.createHash`
+  * https://nodejs.org/api/crypto.html#crypto_crypto_createhash_algorithm_options
+  * @param {string} filepath - filepath of file to checksum
+  * @returns {Promise} checksum value calculated from file
+  **/
+  async _hash(algorithm, filepath) {
+    const options = { algorithm };
 
     return new Promise((resolve, reject) =>
-      checksum.file(tempFile, options, (err, sum) => {
+      checksum.file(filepath, options, (err, sum) => {
         if (err) return reject(err);
         return resolve(sum);
-      })
-    );
+      }));
   }
 
-  async enableDuplicateHandling(bucket) {
+  /**
+   * Enable versioning on an s3 bucket
+   *
+   * @param {string} bucket - s3 bucket name
+   * @returns {Promise} promise that resolves when bucket versioning is enabled
```

Comment:
You removed the return so it's not return anything.

Reply:
It still returns a promise even there is no `return` because it is an async function

Reply:
Good catch.

Reply:
Ahhh so much to learn!